x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowInfoAction.java

```diff
@@ -65,19 +65,28 @@ protected void masterOperation(FollowInfoAction.Request request,
         List<String> concreteFollowerIndices = Arrays.asList(indexNameExpressionResolver.concreteIndexNames(state,
             IndicesOptions.STRICT_EXPAND_OPEN_CLOSED, request.getFollowerIndices()));
 
+        List<FollowerInfo> followerInfos = getFollowInfos(concreteFollowerIndices, state);
+        listener.onResponse(new FollowInfoAction.Response(followerInfos));
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(FollowInfoAction.Request request, ClusterState state) {
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
+    }
 
+    static List<FollowerInfo> getFollowInfos(List<String> concreteFollowerIndices, ClusterState state) {
         List<FollowerInfo> followerInfos = new ArrayList<>();
         PersistentTasksCustomMetaData persistentTasks = state.metaData().custom(PersistentTasksCustomMetaData.TYPE);
 
-        for (IndexMetaData indexMetaData : state.metaData()) {
+        for (String index : concreteFollowerIndices) {
+            IndexMetaData indexMetaData = state.metaData().index(index);
             Map<String, String> ccrCustomData = indexMetaData.getCustomData(Ccr.CCR_CUSTOM_METADATA_KEY);
             if (ccrCustomData != null) {
                 Optional<ShardFollowTask> result;
                 if (persistentTasks != null) {
                     result = persistentTasks.taskMap().values().stream()
                         .map(persistentTask -> (ShardFollowTask) persistentTask.getParams())
-                        .filter(shardFollowTask -> concreteFollowerIndices.isEmpty() ||
-                            concreteFollowerIndices.contains(shardFollowTask.getFollowShardId().getIndexName()))
+                        .filter(shardFollowTask -> index.equals(shardFollowTask.getFollowShardId().getIndexName()))
```

Comment:
I think we need to filter ShardFollowTask task before casting. Maybe use `persistentTasks.findTasks` API.

I will make a PR to inject other persistent tasks randomly in our IT tests.

Reply:
Good catch! I will use that method instead.

> I will make a PR to inject other persistent tasks randomly in our IT tests.

:+1: